blog.md

```diff
@@ -1,26 +1,78 @@
+
+As I wrote this I discovered that while figuring out the right combination of tools to run was a fun process, the really interesting stuff was building the actual CI/CD pipeline to deploy and run the application itself. This means that while I'll briefly describe the underlying infrastructure I will not be providing a detailed installation guide.
 
 ## Infrastructure
 
-1. [OpenStack](http://www.openstack.org/) was installed across a set of physical machines using [Ursula](https://github.com/blueboxgroup/ursula).
-2. A 3 node [DEIS](http://deis.io/)
- cluster was then installed on [OpenStack](http://www.openstack.org/) using the terraform instructions [here](https://github.com/paulczar/deis/tree/openstack_provision_script/contrib/openstack).
-3. A standalone coreos node was created using [Terraform](https://terraform.io/) 
-4. A self replicating database was deployed on the [DEIS](http://deis.io/) nodes using the [paulczar/percona-galera](https://github.com/paulczar/docker-percona_galera) docker image which uses [etcd](https://coreos.com/etcd/) to find each other and form a cluster.
-5. A mysql proxy was deployed on the standalone node using the [paulczar/maxscale](https://hub.docker.com/r/paulczar/maxscale/) image which discovers the databases via [etcd](https://coreos.com/etcd/).
-6. [Jenkins](http://jenkins-ci.org/) was deployed to standalone node using the standard [jenkins docker image](https://hub.docker.com/_/jenkins/).
+While an IaaS is not strictly necessary here (I could run Deis straight on bare metal) it makes sense to use something like [OpenStack](http://www.openstack.org/) as provides excellent APIs to spin up VMs.  I installed OpenStack across across a set of physical machines using [Blue Box's](https://www.blueboxcloud.com/) [Ursula](https://github.com/blueboxgroup/ursula).
+
+Next the PaaS itself, I have familiarity with [Deis](http://Deis.io/) already and I really like its (Heroku-esque) user experience.  I span up a three node Deis cluster on OpenStack using the terraform instructions [here](https://github.com/paulczar/Deis/tree/openstack_provision_script/contrib/openstack).
+
+I also span an additional three [CoreOS](https://coreos.com) nodes using [Terraform](https://terraform.io/) on which I ran [Jenkins](http://jenkins-ci.org/) using the standard [jenkins docker image](https://hub.docker.com/_/jenkins/).
+
+To demonstrate a backing service for my application I also ran a Percona database cluster (one on each of the CoreOS nodes) using the [paulczar/percona-galera](https://github.com/paulczar/docker-percona_galera) docker image which uses [etcd](https://coreos.com/etcd/) to find each other and form automatically create the galera replication set. I also deployed a mysql proxy using the [paulczar/maxscale](https://hub.docker.com/r/paulczar/maxscale/) image which discovers the databases via [etcd](https://coreos.com/etcd/).
```

Comment:
This is another run-on sentence; however, instead of just fixing it mechanically, I'd like to see it re-jiggered it a little bit. In particular, you state that your intention is to demonstrate - but this entire article is already a demonstration, so that goes without saying. Consider something like:
« Finally, there is a three-node Percona database cluster running on the CoreOS nodes, itself fronted by a mysql proxy, both of which use etcd for auto-discovery. Docker images are available for both the [cluster] and the [proxy]. »
* The content has been trimmed down since it was superfluous; recall that this article isn't about the infra anyway. :wink: